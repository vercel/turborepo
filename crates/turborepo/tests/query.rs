mod common;

#[cfg(not(windows))]
#[test]
fn test_double_symlink() -> Result<(), anyhow::Error> {
    check_json!(
        "oxc_repro",
        "npm@10.5.0",
        "query",
        "get_dependencies" => "query {
                 file(path: \"./index.js\") {
                    path
                    dependencies {
                      files { items { path } }
                      errors { items { message import } }
                    }
                 }
              }",
    );
    Ok(())
}

#[test]
fn test_trace() -> Result<(), anyhow::Error> {
    // Separate because the `\\` -> `/` filter isn't compatible with ast
    check_json!(
        "turbo_trace",
        "npm@10.5.0",
        "query",
        "get `main.ts` with ast" => "query { file(path: \"main.ts\") { path ast } }",
    );

    insta::with_settings!({ filters => vec![(r"\\\\", "/")]}, {
        check_json!(
            "turbo_trace",
            "npm@10.5.0",
            "query",
            "get `main.ts`" => "query { file(path: \"main.ts\") { path } }",
            "get `main.ts` with dependencies" => "query { file(path: \"main.ts\") { path, dependencies { files { items { path } } } } }",
            "get `button.tsx` with dependencies" => "query { file(path: \"button.tsx\") { path, dependencies { files { items { path } } } } }",
            "get `circular.ts` with dependencies" => "query { file(path: \"circular.ts\") { path dependencies { files { items { path } } } } }",
            "get `invalid.ts` with dependencies" => "query { file(path: \"invalid.ts\") { path dependencies { files { items { path } } errors { items { message } } } } }",
            "get `main.ts` with depth = 0" => "query { file(path: \"main.ts\") { path dependencies(depth: 1) { files { items { path } } } } }",
        );

        Ok(())
    })
}
