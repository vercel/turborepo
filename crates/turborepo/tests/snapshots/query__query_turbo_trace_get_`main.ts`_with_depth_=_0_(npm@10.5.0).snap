---
source: crates/turborepo/tests/query.rs
expression: query_output
---
{
  "data": {
    "file": {
      "path": "main.ts",
      "dependencies": {
        "files": {
          "items": [
            {
              "path": "button.tsx"
            },
            {
              "path": "foo.js"
            },
            {
              "path": "node_modules/repeat-string/index.js"
            }
          ]
        }
      }
    }
  }
}
